frontend-react/src/pages/submissions/SubmissionTable.tsx

```diff
@@ -1,105 +1,86 @@
+import useCursorManager from "../../hooks/UseCursorManager";
+import SubmissionsResource from "../../resources/SubmissionsResource";
+import { getStoredOrg } from "../../contexts/SessionStorageTools";
+import Table, { ColumnConfig, TableConfig } from "../../components/Table/Table";
 
-/* Handles pagination button logic and display */
-function PaginationButtons({ paginator }: { paginator: PaginationController }) {
-    return (
-        <ButtonGroup type="segmented" className="float-right margin-top-5">
-            {paginator.hasPrev && (
-                <Button
-                    type="button"
-                    onClick={() =>
-                        paginator.changeCursor(paginator.currentIndex - 1)
-                    }
-                >
-                    <span>
-                        <IconNavigateBefore className="text-middle" />
-                        Previous
-                    </span>
-                </Button>
-            )}
-            {paginator.hasNext && (
-                <Button
-                    type="button"
-                    onClick={() =>
-                        paginator.changeCursor(paginator.currentIndex + 1)
-                    }
-                >
-                    <span>
-                        Next
-                        <IconNavigateNext className="text-middle" />
-                    </span>
-                </Button>
-            )}
-        </ButtonGroup>
-    );
-}
+import SubmissionFilters from "./SubmissionFilters";
 
 function SubmissionTable() {
-    /* The one-stop-shop for all things filters, pagination, and API responses! */
-    const { filters, paginator, contents } = useContext(
-        SubmissionFilterContext
+    const filterManager = useFilterManager({
+        sort: {
+            column: "timestamp",
+            order: "DESC",
+        },
+    });
+    const cursorManager = useCursorManager();
+
+    /* Our API call! Updates when any of the given state variables update */
+    const submissions: SubmissionsResource[] = useResource(
+        SubmissionsResource.list(),
+        {
+            organization: getStoredOrg(),
+            cursor: cursorManager.values.cursor,
+            endCursor: filterManager.filters.endRange,
+            pageSize: filterManager.filters.pageSize + 1, // Pulls +1 to check for next page
+            sort: filterManager.filters.sort.order,
+            showFailed: false, // No plans for this to be set to true
+        }
     );
 
-    /* Handles pagination button logic and display */
+    /* Effect to add next cursor whenever submissions returns a new array */
+    useEffect(() => {
+        const nextCursor =
+            submissions[filterManager.filters.pageSize]?.timestamp || undefined;
+
+        // Ensures first page cursor is always the start of your range
+        if (cursorManager.values.currentIndex === 0) {
+            cursorManager.controller.reset(filterManager.filters.startRange);
```

Comment:
There's some weird logic in here about swapping sort order and whether to use `startRange` or `endRange` that should be looked into. This might also manifest in new logic inside the `filterManager` to account for this in API calls by swapping the use of `startRange` and `endRange`.

Reply:
Okay, so...

When sort order is DESC, the API will seek anything from the given cursor to the historic infinity, or as far back in time as we dictate with the endRange value. This means, when paginating, the cursor must reflect the changes needed to bring startRange closer to the endRange while endRange stays static. 

That change looks like this:
Present `<[----------]>` Past
Present `<---[-------]>` Past

When sort order is ASC, the opposite is true. The API needs the endRange date to be changed while the startRange, the date closest to present day, stays static.

Present `<[----------]>` Past
Present `<[-------]--->` Past

The API is defining a `between` clause in the database access layer if both dates are provided. For consistency, it's best if we _always_ have both dates set and delivered with the query. Currently I've set one to `new Date()` and the other to `2020-01-01T00:00:00.000Z`. The work that has to be done is:

 - [ ] Ensure the FilterManager is properly setting the range dates, with `startRange` always being newer than `endRange`
 - [ ] Ensure the cursor manager can work independent of the sorting order. The cursor manager should ONLY care about updating the map of cursors with the next cursor, and navigating to that cursor.
 - [ ] FilterManager needs to use the cursor value to update `endRange` when in `ASC` order, and `startRange` when in `DESC` order